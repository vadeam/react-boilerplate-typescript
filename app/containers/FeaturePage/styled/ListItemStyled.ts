import styled from 'styles/styled-components'

export const ListItemStyled = styled.li`
  margin: 1em 0;
`
