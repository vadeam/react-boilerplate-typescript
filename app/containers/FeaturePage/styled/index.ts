export * from './ListStyled'
export * from './ListItemStyled'
export * from './ListItemTitleStyled'
