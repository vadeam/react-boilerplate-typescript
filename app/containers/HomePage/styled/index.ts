export * from './InputStyled'
export * from './FormStyled'
export * from './SectionStyled'
export * from './DisclaimerStyled'
export * from './CurrencyBlockStyled'
export * from './AtPrefixStyled'
export * from './ContentWrapperStyled'
export * from './FrankfurterLabelStyled'
