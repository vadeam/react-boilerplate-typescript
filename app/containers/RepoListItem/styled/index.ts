export * from './IssueIconStyled'
export * from './IssueLinkStyled'
export * from './RepoLinkStyled'
export * from './WrapperStyled'
