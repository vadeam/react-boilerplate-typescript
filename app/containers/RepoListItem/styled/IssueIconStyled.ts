import NormalIssueIcon from 'components/IssueIcon'
import styled from 'styles/styled-components'

export const IssueIconStyled = styled(NormalIssueIcon)`
  fill: #ccc;
  margin-right: 0.25em;
`
