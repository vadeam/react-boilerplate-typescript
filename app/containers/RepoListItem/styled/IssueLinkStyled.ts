import NormalA from 'components/A'
import styled from 'styles/styled-components'

export const IssueLinkStyled = styled(NormalA)`
  height: 100%;
  color: black;
  display: flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
`
