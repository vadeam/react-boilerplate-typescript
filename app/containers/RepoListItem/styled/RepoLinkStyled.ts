import NormalA from 'components/A'
import styled from 'styles/styled-components'

export const RepoLinkStyled = styled(NormalA)`
  height: 100%;
  color: black;
  display: flex;
  align-items: center;
  width: 100%;
`
