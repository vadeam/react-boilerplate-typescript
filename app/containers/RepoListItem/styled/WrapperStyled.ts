import styled from 'styles/styled-components'

export const WrapperStyled = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: space-between;
`
