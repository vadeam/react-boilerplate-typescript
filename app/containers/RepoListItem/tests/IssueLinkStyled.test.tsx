import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import { IssueLinkStyled } from '../styled'

const renderComponent = (props = {}) => {
  const text = 'Test'
  const utils = render(<IssueLinkStyled {...props}>{text}</IssueLinkStyled>)
  const element = utils.queryByText(text)
  return { ...utils, element }
}

describe('<IssueLinkStyled />', () => {
  it('should match the snapshot', () => {
    const { element } = renderComponent()
    expect(element).toMatchSnapshot()
  })

  it('should have a className attribute', () => {
    const { element } = renderComponent()
    expect(element).toHaveAttribute('class')
  })

  it('should adopt a valid attribute', () => {
    const id = 'test'
    const { element } = renderComponent({ id })
    expect(element).toHaveAttribute('id', id)
  })

  it('should not adopt an invalid attribute', () => {
    const { element } = renderComponent({ attribute: 'test' })
    expect(element).not.toHaveAttribute('attribute')
  })
})
