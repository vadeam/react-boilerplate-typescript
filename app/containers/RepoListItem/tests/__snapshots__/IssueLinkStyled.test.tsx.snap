// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueLinkStyled /> should match the snapshot 1`] = `
.c0 {
  color: #41addd;
}

.c0:hover {
  color: #6cc0e5;
}

.c1 {
  height: 100%;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
}

<a
  class="c0 c1"
>
  Test
</a>
`;
